### PR TITLE
fix: remove usage of probability field

### LINKED ISSUE #5735: [Timebox] Remove "Probability" Field from Opportunities
# Current Behavior

We have a "Probability" field on `Opportunities`.

CleanShot 2024-06-04 at 15 44 34

# Desired Behavior

Remove this field as it's not useful.

### PR DESCRIPTION
**Pull request by siiddhantt**
- fixes #5735

### CODE REVIEW

**Comment by lucasbordeau** [comment]
@Bonapara Should we remove the entire probability type from Twenty or just this particular field on opportunity ?

**Comment by Bonapara** [comment]
@lucasbordeau we should keep the `Rating` field type but remove this probability field from the opportunity object.

**Comment by siiddhantt** [comment]
hi!
made changes so the probability field is only removed from the front and server in opportunity, and also the rating field is already present there

**Comment by lucasbordeau** [comment]
@siiddhantt Hi, please could you keep everything related to this field type, we don't want to remove the probability field type, which is a backend metadata feature level, we just want to remove a particular standard field which is on the standard seed data level.

I'll explain in more details what we want here : 
- Keep everything related to field types rating and probability
- Remove this field from the seeds of standard field we have in the code, so it doesn't get created on new Twenty installations
- Propose a way to migrate properly this field if it already exists in a workspace to a custom field, so it will be the same for users who use this field, except this field will turn into a custom field.

For the migration, you might want to test it on your local install, and we will test it on our side before running it for all workspaces. 

This is a bit complex, don't hesitate to ask for more details and you can reach us on our Discord also.

**Comment by charlesBochet** [comment]
@siiddhantt @lucasbordeau Taking a look at this one:

As Lucas said there are different aspects in this work:
- Field types: we should keep RATING type and remove PROBABILITY type which is not used actually.
- Deprecate Field probability on opportunity: we actually don't want to remove it (as users could have data stored in it) but deprecate it. This is quite complex and @ijreilly will be leading the effort here next week. We can remove seeding logic though

Could you restrict the scope of this PR to removing PROBABILITY field type + stop seeding opportunity probability field?
I'm adding specific comments on the PR to guide you

File: packages/twenty-front/src/modules/object-metadata/utils/formatFieldMetadataItemsAsFilterDefinitions.ts [hunk 1]
```diff
@@ -29,11 +29,6 @@ export const formatFieldMetadataItemsAsFilterDefinitions = ({
       return acc;
     }
 
-    // Todo: remove once Rating fieldtype is implemented
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-metadata/utils/formatFieldMetadataItemsAsFilterDefinitions.ts:32]: this is fine as we will deprecate the field, let's keep it

File: packages/twenty-front/src/modules/object-metadata/utils/getObjectMetadataItemsMock.ts [hunk 2]
```diff
@@ -466,23 +466,6 @@ export const getObjectMetadataItemsMock = () => {
           fromRelationMetadata: null,
           toRelationMetadata: null,
         },
-        {
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-metadata/utils/getObjectMetadataItemsMock.ts:469]: ok :)

File: packages/twenty-front/src/generated/graphql.tsx [hunk 3]
```diff
@@ -233,7 +233,6 @@ export enum FieldMetadataType {
   Numeric = 'NUMERIC',
   Phone = 'PHONE',
   Position = 'POSITION',
-  Probability = 'PROBABILITY',
```
Comment (charlesBochet) [inline packages/twenty-front/src/generated/graphql.tsx:236]: ok :)

File: packages/twenty-front/src/generated-metadata/graphql.ts [hunk 4]
```diff
@@ -327,7 +327,6 @@ export enum FieldMetadataType {
   Numeric = 'NUMERIC',
   Phone = 'PHONE',
   Position = 'POSITION',
-  Probability = 'PROBABILITY',
```
Comment (charlesBochet) [inline packages/twenty-front/src/generated-metadata/graphql.ts:330]: ok :)

File: packages/twenty-front/src/modules/object-record/hooks/__mocks__/useFindManyRecords.ts [hunk 5]
```diff
@@ -53,7 +52,6 @@ export const query = gql`
                 updatedAt
                 companyId
                 stage
-                probability
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/hooks/__mocks__/useFindManyRecords.ts:56]: ok :) as this is mocked data

File: packages/twenty-front/src/modules/object-record/record-field/hooks/usePersistField.ts [hunk 6]
```diff
@@ -87,9 +85,6 @@ export const usePersistField = () => {
           isFieldBoolean(fieldDefinition) &&
           isFieldBooleanValue(valueToPersist);
 
-        const fieldIsProbability =
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/hooks/usePersistField.ts:90]: let's rename this to fieldIsRating and use it in isValuePersistable

File: packages/twenty-front/src/modules/object-record/record-field/types/guards/assertFieldMetadata.ts [hunk 7]
```diff
@@ -51,19 +51,17 @@ type AssertFieldMetadataFunction = <
                           ? FieldNumberMetadata
                           : E extends 'PHONE'
                             ? FieldPhoneMetadata
-                            : E extends 'PROBABILITY'
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/object-record/record-field/types/guards/assertFieldMetadata.ts:54]: let's rename 'PROBABILITY' to 'RATING' here but keep it

File: packages/twenty-front/src/modules/search/hooks/__mocks__/useFilteredSearchEntityQuery.ts [hunk 8]
```diff
@@ -25,7 +25,6 @@ export const query = gql`
                 updatedAt
                 companyId
                 stage
-                probability
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/search/hooks/__mocks__/useFilteredSearchEntityQuery.ts:28]: ok

File: packages/twenty-front/src/modules/search/hooks/__mocks__/useFilteredSearchEntityQuery.ts [hunk 9]
```diff
@@ -50,7 +49,6 @@ export const query = gql`
...
```
Comment (charlesBochet) [inline packages/twenty-front/src/modules/search/hooks/__mocks__/useFilteredSearchEntityQuery.ts:53]: ok

File: packages/twenty-front/src/modules/object-record/record-field/types/guards/assertFieldMetadata.ts (hunk 7 above)
Comment (siiddhantt) [reply packages/twenty-front/src/modules/object-record/record-field/types/guards/assertFieldMetadata.ts:54]: there's already a 'RATING' just a little above

**Comment by FelixMalfait** [comment]
cc @ijreilly could you please take over this PR as it's related to yours? 

Also fyi https://github.com/twentyhq/twenty/issues/3531 we do want to remove the probability field type.
We have the rating field type which is good enough to represent this. Eventually at some point we'll have display options on the rating field type "stars, progress bar, etc."

**Comment by siiddhantt** [comment]
Hi @ijreilly thanks for the update! I appreciate your help on this.
Since you've been working on a related issue, you could take over this. Thanks :)

**Comment by ijreilly** [comment]
Got it, @siiddhantt thanks a lot for your contribution and interest in twenty!

**Review by ijreilly — APPROVED** [review]

**Comment by ijreilly** [comment]
Closes #6259

**Pull request merged**